### ISSUE TITLE
Adding an empty data values to a library removes that libraries' data values

### ISSUE THREAD

**Issue by cari-lynn** [k14s/ytt#423]
**What steps did you take:**
Adding empty data values to a library with existing data values is successful on v0.31.0 but when updating to v0.32.0, it now reports an error.

```
.
├── config.yml
└── _ytt_lib/
     └── library/
          └── config.yml
          └── values.yml
```

`config.yml`
```
#@ load("@ytt:data", "data")
#@ load("@ytt:library", "library")
#@ load("@ytt:template", "template")

#@ library = library.get("library")
--- #@ template.replace(library.with_data_values(data.values).eval())
```

`_ytt_lib/library/config.yml`
```
#@ load("@ytt:data","data")
---
dvs: #@ data.values.example
```

`_ytt_lib/library/values.yml`
```
#@data/values
---
example: I should appear in the output of the library
```
**What happened:**
(v0.32.0+) When the data values that are defined in the library are used in that library, ytt cannot find them.
```
ytt: Error:
- library.eval: Evaluating library 'library':
    in <toplevel>
      config.yml:10 | --- #@ template.replace(library.with_data_values(data.values).eval())

    reason:
     - NoneType has no .example field or method
         in <toplevel>
           _ytt_lib/library/config.yml:3 | dvs: #@ data.values.example
```
**What did you expect:**
Just like what happens when using v0.31.0, I expect this to succeed. 
```
dvs: I should appear in the output of the library
```

**Anything else you would like to add:**
In v0.31.0 the type of an empty `data.values` is `struct`. In v0.32.0+ it is `NoneType`. When adding a `NoneType` data values to the library via `with_data_values()`, it is interfering with the data values that already exist for the library.

This is happening because we overlay a `nil` data value Document onto the existing data values, essentially wiping the existing data values.
https://github.com/vmware-tanzu/carvel-ytt/blob/develop/pkg/workspace/data_values_pre_processing.go#L63
A possible solution could be to check if a data values value is nil prior to overlaying or adding it to the list of data values.

**Environment:**

- ytt version (use `ytt --version`): v0.32.0+
- OS (e.g. from `/etc/os-release`):

---
Vote on this request

This is an invitation to the community to vote on issues, to help us prioritize our backlog. Use the "smiley face" up to the right of this comment to vote.

👍 "I would like to see this addressed as soon as possible"
👎 "There are other more important things to focus on right now"

We are also happy to receive and review Pull Requests if you want to help working on this issue.

**Comment by pivotaljohn** [comment]
Thanks for the succinct reproduction, @cari-lynn, this is very clear. 👍🏻 

This is undesirable behavior. I'm wracking my head to think of a case where we'd want Data Values to be actually `null`/`None` and not just empty. Gonna chew on that, myself.

In the meantime, this is decidedly a regression ==> "carvel accepted"
And you reproduced the defect ==> "can be replicated"

**Comment by pivotaljohn** [comment]
This is a regression with known impact. There's a work around (where authors can defensively perform a null/None check), but we don't want that to proliferate.

Let's get this in for v0.35.0 ==> "Prioritized Backlog"

**Comment by cari-lynn** [comment]
The fix for this will be available in the next release v0.35.0. An empty data values is a `struct` once again rather than `None`.

**Comment by aaronshurley** [comment]
[v0.35.0](https://github.com/vmware-tanzu/carvel-ytt/releases/tag/v0.35.0) was released a few days ago. Closing out this issue.
